### PR TITLE
Introduce default dark theme

### PR DESCRIPTION
**Pull request by adguernier**
## Problem

React-admin offers both a light and dark theme, but since only one theme is enabled by defaumt, the theme switcher doesn't appear in new admins. 

## Solution

Update the `<Admin>` component to include dark theme by default. This enables the theme switcher on all apps.

## TODO
- [x] Add the defaultDarkTheme in AdminUI
- [x] Remove the default darkTheme in stories
- [x] Update documentation
- [x] Add upgrade guide

### CODE REVIEW

File: docs/Admin.md [hunk 1]
```diff
@@ -433,29 +434,48 @@ const App = () => (
 
 ## `darkTheme`
 
-If you want to support both light and dark mode, you can provide a `darkTheme` in addition to the `theme` prop. The app will use the `darkTheme` by default for users who prefer the dark mode at the OS level, and users will be able to switch from light to dark mode using a new app bar button leveraging [the `<ToggleThemeButton>` component](./ToggleThemeButton.md).
+React-admin provides a [built-in dark theme by default](./AppTheme.md#default). The app will use the `darkTheme` by default for users who prefer the dark mode at the OS level, and users will be able to switch from light to dark mode using an app bar button leveraging [the `<ToggleThemeButton>` component](./ToggleThemeButton.md).
```
Comment (djhi) [inline docs/Admin.md:437]:
```suggestion
React-admin provides a [built-in dark theme](./AppTheme.md#default). The app will use the `darkTheme` by default for users who prefer the dark mode at the OS level, and users will be able to switch from light to dark mode using [the `<ToggleThemeButton>` component](./ToggleThemeButton.md).
```

File: docs/Admin.md [hunk 2]
```diff
@@ -433,29 +434,48 @@ const App = () => (
...
 
 <video controls autoplay muted loop>
   <source src="./img/ToggleThemeButton.webm" type="video/webm"/>
   Your browser does not support the video tag.
 </video>
 
+If you want to override it, you can provide your custom `darkTheme` in addition to the `theme` prop:
```
Comment (djhi) [inline docs/Admin.md:444]:
```suggestion
If you want to override it, you can provide your own `darkTheme` in addition to the `theme` prop:
```

File: docs/Admin.md [hunk 3]
```diff
@@ -600,6 +620,36 @@ Refer to each layout component documentation to understand the props it accepts.
 
 Finally, you can also pass a custom component as the `layout` prop. It must contain a `{children}` placeholder, where react-admin will render the page content. Check [the custom layout documentation](./Layout.md#writing-a-layout-from-scratch) for examples, and use the [default `<Layout>`](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/layout/Layout.tsx) as a starting point.
 
+## `lightTheme`
+
+React-admin provides a [built-in light theme by default](./AppTheme.md#default). The app will use the `lightTheme` by default for users who prefer the light mode at the OS / browser level, and users will be able to switch from light to dark mode using an app bar button leveraging [the `<ToggleThemeButton>` component](./ToggleThemeButton.md) if the `darkTheme` prop is provided or leaved by default.
```
Comment (djhi) [inline docs/Admin.md:625]:
```suggestion
React-admin provides a [built-in light theme](./AppTheme.md#default). The app will use the `lightTheme` by default for users who prefer the light mode at the OS / browser level, and users will be able to switch from light to dark mode using [the `<ToggleThemeButton>` component](./ToggleThemeButton.md) if the `darkTheme` prop is provided or leaved by default.
```

File: docs/Admin.md [hunk 4]
```diff
@@ -600,6 +620,36 @@ Refer to each layout component documentation to understand the props it accepts.
...
+
+<video controls autoplay muted loop>
+  <source src="./img/ToggleThemeButton.webm" type="video/webm"/>
+  Your browser does not support the video tag.
+</video>
+
+If you want to override it, you can provide your custom `lightTheme`:
```
Comment (djhi) [inline docs/Admin.md:632]:
```suggestion
If you want to override it, you can provide your own `lightTheme`:
```

File: docs/AppTheme.md [hunk 5]
```diff
@@ -60,14 +60,17 @@ It's a common practice to support both a light theme and a dark theme in an appl
   Your browser does not support the video tag.
 </video>
 
+React-admin provides a [built-in dark theme by default](#default), the default application theme depends on the user's system settings. If the user has chosen a dark mode in their OS, react-admin will use the dark theme. Otherwise, it will use the light theme.
 
-React-admin's `<Admin>` component accepts a `darkTheme` prop in addition to the `theme` prop. 
+In addition, users can switch from one theme to the other using [the `<ToggleThemeButton>` component](./ToggleThemeButton.md), which appears in the AppBar as soon as you define a `darkTheme` prop.
```
Comment (djhi) [inline docs/AppTheme.md:65]:
```suggestion
In addition, users can switch from one theme to the other using [the `<ToggleThemeButton>` component](./ToggleThemeButton.md), that appears in the AppBar as soon as you define a `darkTheme` prop.
```

File: docs/ToggleThemeButton.md [hunk 6]
```diff
@@ -13,7 +13,7 @@ The `<ToggleThemeButton>` component lets users switch from light to dark mode, a
   Your browser does not support the video tag.
 </video>
 
-It is enabled by default in the `<AppBar>` as soon as you define a dark theme via [the `<Admin darkTheme>` prop](./Admin.md#darktheme).
+It is enabled by default in the `<AppBar>` as soon React-admin provides a [built-in dark theme by default](./AppTheme.md#default).
```
Comment (djhi) [inline docs/ToggleThemeButton.md:16]:
```suggestion
It is enabled by default in the `<AppBar>` as React-admin provides a [built-in dark theme](./AppTheme.md#default).
```

File: docs/Admin.md [hunk 7]
```diff
@@ -843,7 +893,15 @@ const App = () => (
...
 
-If you want to support both a light and a dark theme, check out [the `<Admin darkTheme>` prop](#darktheme). 
+React-admin provides a [built-in dark theme by default](./AppTheme.md#default). If you want to override it, check out [the `<Admin darkTheme>` prop](#darktheme). 
+
+### Theme Priorities
+
+The following priorities are applied depending on whether the `theme` prop is provided or not:
+
+- if `theme` is not provided, the default light and dark theme are used;
+- if `theme` is not provided the `lightTheme` prop is ignored;
```
Comment (djhi) [inline docs/Admin.md:903]: Are you sure?

File: docs/ToggleThemeButton.md [hunk 8]
```diff
@@ -71,7 +84,9 @@ export const MyAppBar = () => (
 
 ## Creating A Dark Theme
 
-For this button to work, you must provide a dark theme to the `<Admin>` component. The `darkTheme` should be a JSON object that follows the [Material UI theme specification](https://material-ui.com/customization/theming/).
+For this button to work, you must provide a dark theme to the `<Admin>` component. React-admin provides a [built-in dark theme by default](./AppTheme.md#default), but you can override it according to your needs.
```
Comment (djhi) [inline docs/ToggleThemeButton.md:87]:
```suggestion
For this button to work, you must provide a dark theme to the `<Admin>` component. React-admin provides a [built-in dark theme](./AppTheme.md#default), but you can override it according to your needs.
```

File: docs/Upgrade.md [hunk 9]
```diff
@@ -18,6 +18,24 @@ If you previously relied on the fact that the rows were not clickable by default
 </Datagrid>
 ```
 
+## A `darkTheme` is now provided by default
+
+If not themes are configured in the `<Admin>` component prop, React-admin enable the [dark mode feature](https://marmelab.com/react-admin/AppTheme.html#light-and-dark-themes) by default.
```
Comment (djhi) [inline docs/Upgrade.md:23]:
```suggestion
If themes are not configured using the `<Admin>` component props, React-admin enables the [dark mode feature](https://marmelab.com/react-admin/AppTheme.html#light-and-dark-themes).
```

File: docs/Upgrade.md [hunk 10]
```diff
@@ -18,6 +18,24 @@ If you previously relied on the fact that the rows were not clickable by default
...
+
+If you don't need the dark mode feature, you need to explicitly disable it:
```
Comment (djhi) [inline docs/Upgrade.md:25]:
```suggestion
If you don't need the dark mode feature, you'll have to explicitly disable it:
```

File: docs/Upgrade.md [hunk 11]
```diff
@@ -18,6 +18,24 @@ If you previously relied on the fact that the rows were not clickable by default
...
+```diff
+-<Admin>
++<Admin darkTheme={null}>
+   ...
+</Admin>
+```
+
+To best manage the `theme` prop, here are the priorities applied depending on whether the `theme` prop is provided or not:
```
Comment (djhi) [inline docs/Upgrade.md:33]:
```suggestion
Here are the priorities applied depending on whether the `theme` prop is provided or not:
```

File: docs/Upgrade.md [hunk 12]
```diff
@@ -18,6 +18,24 @@ If you previously relied on the fact that the rows were not clickable by default
...
+
+- if `theme` is not provided, the [default light and dark theme](https://marmelab.com/react-admin/AppTheme.html#default) are used;
+- if `theme` is not provided the `lightTheme` prop is ignored (no change from before);
```
Comment (djhi) [inline docs/Upgrade.md:36]: Nope

**Review by djhi — CHANGES_REQUESTED** [review]

File: docs/Admin.md [hunk 13]
```diff
@@ -145,11 +145,12 @@ Here are all the props accepted by the component:
...
 | `dashboard`        | Optional | `Component`    | -              | The content of the dashboard page                        |
-| `darkTheme`        | Optional | `object`       | -              | The dark theme configuration                             |
+| `darkTheme`        | Optional | `object`       | `defaultDarkTheme` (the default built-in dark theme)              | The dark theme configuration                             |
 | `defaultTheme`     | Optional | `boolean`      | `false`        | Flag to default to the light theme                       |
 | `disableTelemetry` | Optional | `boolean`      | `false`        | Set to `true` to disable telemetry collection            |
 | `i18nProvider`     | Optional | `I18NProvider` | -              | The internationalization provider for translations       |
 | `layout`           | Optional | `Component`    | `Layout`       | The content of the layout                                |
+| `lightTheme`        | Optional | `object`       | `defaultLightTheme` (the default built-in light theme)              | The light theme configuration                             |
```
Comment (fzaninotto) [inline docs/Admin.md]: We shouldn't document the lightTheme IMO, it's confusing.

File: docs/Admin.md [hunk 14]
```diff
@@ -145,11 +145,12 @@ Here are all the props accepted by the component:
 | `basename`         | Optional | `string`       | -              | The base path for all URLs                               |
 | `catchAll`         | Optional | `Component`    | `NotFound`     | The fallback component for unknown routes                |
 | `dashboard`        | Optional | `Component`    | -              | The content of the dashboard page                        |
-| `darkTheme`        | Optional | `object`       | -              | The dark theme configuration                             |
+| `darkTheme`        | Optional | `object`       | `defaultDarkTheme` (the default built-in dark theme)              | The dark theme configuration                             |
```
Comment (fzaninotto) [inline docs/Admin.md]: you should also set the default value for the `theme` to `defaultLightTheme`

**Review by fzaninotto — CHANGES_REQUESTED** [review]
Neat!

**Comment by fzaninotto** [comment]
We have many stories that set a custom dark theme. You should revisit these, too. 

image

**Comment by fzaninotto** [comment]
Interesting side effect: My OS defaults to dark theme, and so every story in the Storybook that uses `<Admin>` or `<AdminContext>` defaults to dark theme, too. 

image

I suggest we systematically set the `defaultTheme="light"` in stories to avoid that.

File: docs/Upgrade.md [hunk 15]
```diff
@@ -18,6 +18,28 @@ If you previously relied on the fact that the rows were not clickable by default
...
+If you don't need the dark mode feature, you'll have to explicitly disable it:
+
+```diff
+-<Admin>
++<Admin darkTheme={null}>
+   ...
+</Admin>
+```
+
+If the `theme` prop is defined but not the `darkTheme` prop, theme switcher button in the app bar isn't displayed:
```
Comment (djhi) [inline docs/Upgrade.md:34]:
```suggestion
If the `theme` prop is defined but not the `darkTheme` prop, the theme switcher button in the app bar isn't displayed:
```

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by fzaninotto — APPROVED** [review]

**Review by djhi — APPROVED** [review]

**Pull request merged**